quickwit-search/src/client.rs

```diff
@@ -28,11 +29,26 @@ use tokio_stream::wrappers::UnboundedReceiverStream;
 use tonic::transport::Channel;
 use tonic::transport::Endpoint;
 use tonic::Request;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+use tracing::*;
 
 use crate::error::parse_grpc_error;
 use crate::SearchError;
 use crate::SearchService;
 
+struct MetadataMap<'a>(&'a mut tonic::metadata::MetadataMap);
+
+impl<'a> Injector for MetadataMap<'a> {
+    /// Set a key and value in the MetadataMap.  Does nothing if the key or value are not valid inputs
+    fn set(&mut self, key: &str, value: String) {
+        if let Ok(key) = tonic::metadata::MetadataKey::from_bytes(key.as_bytes()) {
```

Comment:
Variable shadowing is not going to make everybody happy here :) Let' go with something along those lines:
```rust
    fn set(&mut self, key: &str, value: &str) {
        if let Ok(metadata_key) = tonic::metadata::MetadataKey::from_bytes(key.as_bytes()) {
            if let Ok(metadata_value) = tonic::metadata::MetadataValue::from_str(value) {
                self.0.insert(metadata_key, metadata_value);
            }
        }
```

Reply:
indeed I don't like it either.